SWLOR.Game.Server/Service/Item.cs

```diff
@@ -42,6 +43,19 @@ public static void CacheData()
             }
 
             Console.WriteLine($"Loaded {_items.Count} items.");
+
+            // Cache 2da values that we need.  Create a new array for each row, otherwise they
+            // end up pointing to the same array object (and get overwritten).
+            foreach (var baseItem in Enum.GetValues(typeof(BaseItem)).Cast<int>())
+            {
+                string[] values = new string[3];
+                values[0] = Get2DAString("baseitems", "CritThreat", baseItem);
```

Comment:
Can you do the parsing to an int here so it only needs to happen one time rather than on every GetCriticalThreatRange / GetCriticalModifier / GetWeaponSize call?

Reply:
Sure thing.